Std/Data/Array/Lemmas.lean

```diff
@@ -724,3 +724,182 @@ theorem all_eq_true_iff_forall_mem {l : Array α} : l.all p ↔ ∀ x, x ∈ l 
 /-! ### erase -/
 
 @[simp] proof_wanted erase_data [BEq α] {l : Array α} {a : α} : (l.erase a).data = l.data.erase a
+
+/-! ### swap -/
+
+@[simp] theorem swap_def_jeqi {a: Array α} {i j : Nat}
+    (_ : i < a.size := by trivial) (_ : j < a.size := by trivial)
```

Comment:
Why are you writing `by trivial` here? Doesn't seem like a useful autoparam.

Reply:
no problem. .. often times when I am using theorems like this these minor hypothesis can be distracting, so I made them option assumption they have already been proven somewhere before the theorem's use. This is just a personal preference so I will go ahead and remove these optional params